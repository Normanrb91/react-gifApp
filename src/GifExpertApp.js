import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = () => {

    const [categories, setCategories] = useState('');

        return (

            <>
            {
            categories ? (
                <div className="container ">
                    <h2 className="mt-5 mb-3">Gif App</h2>
                    <AddCategory setCategories={setCategories} />
                    <hr />
                    {
                    <GifGrid key={categories} category={categories} />
                    }
                </div>
            ) :
                <div className="container">
                    <h2 className="mt-5 mb-3">Gif App</h2>
                    <AddCategory setCategories={setCategories} />
                    <hr />
                </div>
            }

            </>
        );

}
