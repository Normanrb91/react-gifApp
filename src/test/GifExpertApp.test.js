import React from 'react';

import { shallow } from 'enzyme';
import {GifExpertApp} from '../GifExpertApp'




describe('pruebas con <GifExpertApp />', () => {

    test('debe mostrar el componente correctamente ', () => {

        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
        
    })
    
    
})
