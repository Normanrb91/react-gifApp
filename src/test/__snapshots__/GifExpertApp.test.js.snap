// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con <GifExpertApp /> debe mostrar el componente correctamente  1`] = `
<Fragment>
  <div>
    <h2>
      GifExpertApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
  </div>
</Fragment>
`;
